x-pack/plugins/fleet/server/plugin.ts

```diff
@@ -356,18 +374,32 @@ export class FleetPlugin
               asInternalUser: service.asInternalUser,
             };
           },
-          authz: await getAuthzFromRequest(request),
+          authz,
           epm: {
             // Use a lazy getter to avoid constructing this client when not used by a request handler
             get internalSoClient() {
-              return appContextService
-                .getSavedObjects()
-                .getScopedClient(request, { excludedExtensions: [SECURITY_EXTENSION_ID] });
+              return getInternalSoClient();
             },
           },
           get spaceId() {
             return deps.spaces.spacesService.getSpaceId(request);
           },
+
+          async getSoClient() {
+            if (routeAuthz && routeAuthz.granted) {
+              return {
+                client: routeAuthz.grantedByFleetPrivileges
```

Comment:
I am wondering if there is any value of not always using the internal SO client

Reply:
🤔 it probably would not hurt to always use the internal client. If that was to be the case, the we likely would not need this property on the Route context. I have not spent time thinking about it, but if you feel it is a safe change, we make it in this PR.  Let us know

Reply:
Yes I think it's safe change to make, and it make things simpler to use the same saved object client.

Reply:
done 2886e3dc4a2f459c0e515537c5a01fd2556c96fb

Reply:
@juliaElastic ,

Re: 

> Is the agent policy API going to filter out the package policies that do not belong to `endpoint` package?

I'm not sure yet about how this one should be implemented and it warrants some discussions. Our Endpoint use cases don't need `update/create/delete` access, so for us, it would be ok to filter out any package policy that does not match the authz package names. But for cases (in the future or for other integrations) where Update/Delete is needed, that gets more complicated.

@ashokaditya,
we need to explore each one of these individually and see if we can come up with good suggestions on the approach. We should use the tracking issues you created to expand upon this.

Reply:
That's a good point, I think the problem is that the Agent policy is a resource that doesn't belong to one package only. 
We don't have to solve now for use cases that require edit access.

Reply:
Last comment by Julia above is for this comment/response by me here:

https://github.com/elastic/kibana/pull/145361#issuecomment-1348690980

_(Sorry. Coffee had not kicked in yet, and I initially posted by response in the wrong thread and then correct it)_